scripts/jenkins/Jenkinsfile.release

```diff
@@ -0,0 +1,326 @@
+    def dockerArgs = ""
+    if (opts.contains(DOCKER_OPTS.MOUNT_MIRROR)) {
+      dockerArgs += "-v ${env.HOME}/git_mirrors:/home/jenkins/git_mirrors "
+    }
+    if (opts.contains(DOCKER_OPTS.PTRACE)) {
+      dockerArgs += "--cap-add SYS_PTRACE "
+    }
+    docker.withRegistry("https://${REGISTRY}",
+                        'docker-hub-elektra-jenkins') {
+      timeout(activity: true, time: 40, unit: 'MINUTES') {
+        def cpu_count = cpuCount()
+        withEnv(["MAKEFLAGS='-j${cpu_count+2} -l${cpu_count*2}'",
+                 "CTEST_PARALLEL_LEVEL='${cpu_count+2}'",
+                 "XDG_CONFIG_HOME=${WORKSPACE}/xdg/user",
+                 "XDG_CONFIG_DIRS=${WORKSPACE}/xdg/system"]) {
+          echo "Starting ${STAGE_NAME} on ${NODE_NAME} using ${image.id}"
+          checkout scm
+          docker.image(image.id)
+                .inside(dockerArgs) { cl() }
+        }
+      }
+    }
+  }
+}
+
+/* Create a new Docker Image description
+ *
+ * @param name Name of the image, will be extended with registry, a common
+ *             prefix and a tag
+ * @param idFun Closure describing how the image id should be formatted
+ *                      (see idTesting() / idWebsite())
+ * @param context Build context for the docker build (base directory that will
+ *                be sent to the docker agent). Relative to the current working
+ *                directory.
+ * @param file Path to Dockerfile relative to the current working directory.
+ * @param autobuild If it should be automatically build at the start of the
+ *                  Jenkins run. If false it can be build manually
+ *                  (see buildImageStage()).
+ */
+def createDockerImageDesc(name, idFun, context, file, autobuild=true) {
+  def prefix = 'build-elektra'
+  def fullName = "${REGISTRY}/${prefix}-${name}"
+  def map = [
+    name: fullName,
+    context: context,
+    file: file,
+    autobuild: autobuild,
+    exists: false
+  ]
+  return idFun(map)
+}
+
+/* Build image ID of docker images used for tests
+ *
+ * We use identifiers in the form of name:yyyyMM-hash
+ * The hash is build from reading the Dockerfile. Hence it needs to be
+ * checked out before it can be calculated.
+ * @param imageMap Map identifying an docker image (see DOCKER_IMAGES)
+ */
+def idTesting(imageMap) {
+  def cs = checksum(imageMap.file)
+  def dateString = dateFormatter(NOW)
+  imageMap.id = "${imageMap.name}:${dateString}-${cs}"
+  return imageMap
+}
+
+
+/* Format the date input
+ * @param date Date to format
+ */
+def dateFormatter(date) {
+  df = new SimpleDateFormat("yyyyMM")
+  return df.format(date)
+}
+
+/* Generate the checksum of a file
+ * @param file File to generate a checksum for
+ */
+def checksum(file) {
+  // Used to identify if a Dockerfile changed
+  // TODO expand to use more than one file if Dockerfile ever depends on
+  //      external files
+  return sh(returnStdout: true,
+            script: "cat $file | sha256sum | dd bs=1 count=64 status=none")
```

Comment:
Why do we cut off here after 64 bytes? I think we can effort to store the whole sha256sum.

Reply:
I think what this does is just truncate the output of the command such that the file name is not appended after the SHA-256 sum. Otherwise I agree that we can store the whole thing.